client/state/posts/selectors.js

```diff
@@ -105,6 +134,13 @@ export const getSitePost = createSelector( ( state, siteId, postId ) => {
  */
 export const getSitePostsForQuery = createSelector(
 	( state, siteId, query ) => {
+		if ( ! siteId ) {
+			// TODO remove me after testing and before merge
+			// eslint-disable-next-line no-console
+			console.warn( 'getSitePostsForQuery called without siteId' );
+			return null;
+		}
+
```

Comment:
It probably makes sense to support all-sites mode in this selector also, and rename it to `getPostsForQuery`.  Separately, it should probably be removed entirely, because its usage is very low and it has incorrect behavior (see TODO comment below).